sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py

```diff
@@ -5,53 +5,94 @@
 from ._shared.avro.avro_io import DatumReader
 
-from ._models import QuickQueryError
+from ._models import BlobQueryError
 
 
-class QuickQueryReader(object):  # pylint: disable=too-many-instance-attributes
-    """A streaming object to read quick query result.
+class BlobQueryReader(object):  # pylint: disable=too-many-instance-attributes
+    """A streaming object to read query results.
 
     :ivar str name:
-        The name of the blob for the quick query request.
+        The name of the blob being quered.
     :ivar str container:
         The name of the container where the blob is.
     :ivar dict response_headers:
         The response_headers of the quick query request.
-    :ivar int total_bytes:
+    :ivar int size:
         The size of the total data in the stream.
+    :ivar int bytes_processed:
+        The number of bytes that the service has so far processed and
+        returned. This value is incrementally updated as the data is streamed.
+    :ivar bytes record_delimiter:
+        The delimiter used to separate lines, or records with the data. The `records`
+        method will return these lines via a generator.
     """
 
     def __init__(
         self,
-        client=None,
         name=None,
         container=None,
-        progress_callback=None,
+        errors='strict',
+        record_delimiter='\n',
         encoding=None,
-        **kwargs
+        headers=None,
+        response=None
     ):
         self.name = name
         self.container = container
-        self.response_headers = None
-        self.total_bytes = None
+        self.response_headers = headers
+        self.size = 0
         self.bytes_processed = 0
-        self._progress_callback = progress_callback
-        self._client = client
-        self._request_options = kwargs
+        self.record_delimiter = record_delimiter
+        self._errors = errors
         self._encoding = encoding
+        self._parsed_results = DataFileReader(QuickQueryStreamer(response), DatumReader())
+        self._first_result = self._process_record(next(self._parsed_results))
 
     def __len__(self):
-        return self.total_bytes
+        return self.size
+
+    def _process_record(self, result):
+        self.size = result.get('totalBytes', self.size)
+        self.bytes_processed = result.get('bytesScanned', self.bytes_processed)
+        if 'data' in result:
+            return result.get('data')
+        if 'fatal' in result:
+            error = BlobQueryError(
+                error=result['name'],
+                is_fatal=result['fatal'],
+                description=result['description'],
+                position=result['position']
+            )
+            if (self._errors == 'ignore' and result['fatal']) or self._errors == 'strict':
+                raise error
+            if self._errors == 'ignore':
+                return None
+            if not self._errors(error):
+                raise error
```

Comment:
I believe we're not supposed to stop processing result on these errors but report them if they happen.
@seanmcc-msft @gapra-msft could you please chime in here as you guys had more insight into feature spec?

Reply:
That is correct, the Quick Query team wants us to continue processing, even if we encounter a fatal error.

Reply:
This is useful info - thanks @seanmcc-msft, @kasobol-msft!
I will revisit the error structure/handling.